ui_framework/test/common_tests.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { render } from 'enzyme';
+
+/**
+ * Ensures the rendered class matches the jest snapshot.
+ * @param reactClass
+ * @param props
+ */
+export function basicRenderTest(reactClass, props = {}) {
```

Comment:
My motivation is that these three basic tests will eventually be duplicated hundreds of times across the ui framework, and it will be painful to add each of them manually, especially given the number of components that will have no props, and we only want to make sure they are rendered, handle the basic html attributes, and can contain children.

I understand your point about readability, but if the same chunk of code is repeated multiple times, it's not giving new or helpful information, it's just noise.  But a smaller, more concise file is easier to read/scan/find relevant information.  The tests aren't interesting, and they will happen for almost every component. The more interesting tests are those that test specific functionality unique to that component.  That is what I'd prefer to focus in on when looking at a test file, not having to first scroll through the basic test functions that I've seen 1000 times before.

If I un-DRY this, I'll turn 8 unique loc into ~50 duplicate loc.  As time grows that difference will only grow. Take for instance how many react components were created in the React Landing Page PR.  Even just the components exported from [this table.js file.](https://github.com/Stacey-Gammon/kibana/blob/5f514b7358a4f7319a8639eb6be5707db5c7e0e6/ui_framework/components/table/table.js).  All in total it seems like there are about 30 components, for one PR, for a landing page. We are looking at 30 * 3 = 60 loc vs 30 *15 = 450 loc.  That will only grow.

gtg so I have to cut my thoughts a bit short now.